### PR TITLE
Use H5Lexists to determine if object exists

### PR DESCRIPTION
**Pull request by brtnfld**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Use H5Lexists to determine if an object exists before trying to get information on the object.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Summit

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/io/hdf/hdf_archive.cpp [hunk 1]
```diff
@@ -162,17 +162,26 @@ bool hdf_archive::is_group(const std::string& aname)
   hid_t p = group_id.empty() ? file_id : group_id.top();
   p       = (aname[0] == '/') ? file_id : p;
 
-  H5O_info_t oinfo;
-  oinfo.type = H5O_TYPE_UNKNOWN;
+  if (H5Lexists(p, aname.c_str(), H5P_DEFAULT) > 0)
+    {
```
Comment (ye-luo) [inline src/io/hdf/hdf_archive.cpp:166]: Please run clang-format on the file.

**Review by ye-luo — APPROVED** [review]

**Review by prckent — APPROVED** [review]
Thanks Scot!

**Comment by prckent** [comment]
Test this please

**Comment by prckent** [comment]
Start testing in-house

**Comment by brtnfld** [comment]
> Thanks Scot!

Sorry I overlooked this in my previous PR.

**Pull request merged**